incubator/validator-core/src/main/java/io/aklivity/zilla/runtime/validator/core/StringEncoding.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.validator.core;
+
+import org.agrona.DirectBuffer;
+
+public enum StringEncoding
+{
+    UTF_8
+    {
+        @Override
+        public boolean validate(
+            DirectBuffer data,
+            int index,
+            int length)
+        {
+            byte[] bytes = new byte[length];
+            data.getBytes(index, bytes);
```

Comment:
Remove these 2 lines and use `DirectBuffer` API to access bytes instead of `bytes[...]` array syntax.